### ISSUE TITLE
Pickup node_modules solidity files

### ISSUE THREAD

**Issue by jchittoda** [intellij-solidity/intellij-solidity#109]
Plugin should pickup solidity packages from node_modules folder. For example openzeppelin code should be picked up from node_modules folder.

**Comment by 0v1se** [comment]
@jchittoda what do you mean by "pickup"?
If you rererence sol files from node_modules folder, plugin will find references correctly.
Pls, give us an example

**Comment by jchittoda** [comment]
Solc does pickup and import the files present in node_module folder like this.

```import "openzeppelin-solidity/contracts/token/ERC20/StandardToken.sol";```

But if I open the .sol file having above line of imports, will show errors as plugin is not able to pick files from node_modules.

**Comment by evgenynacu** [comment]
Give me an example repository if you have it.
For example, here: https://github.com/daonomic/sale/blob/master/contracts/AbstractSale.sol#L4

I import sol file from other package. It works for me

**Comment by 0v1se** [comment]
Give me an example repository if you have it.
For example, here: https://github.com/daonomic/sale/blob/master/contracts/AbstractSale.sol#L4

I import sol file from other package. It works for me

**Comment by jchittoda** [comment]
You can try with this project https://github.com/MainframeHQ/contracts
I am using intellij-Solidity 2.1.2

**Comment by SerCeMan** [comment]
Hi, @jchittoda!

Could you please attach a screenshot of the problem, I tried the project you specified and it worked well for me, 
<img width="1399" alt="screen shot 2018-09-20 at 11 59 23 pm" src="https://user-images.githubusercontent.com/1780970/45859706-5ca27d00-bd31-11e8-9066-caf3f71e4e17.png">

**Comment by jchittoda** [comment]
@SerCeMan Did you tried to compile the Solidity files using "Build" menu -> "Compile Solidity"? Let me know what is the output you see in Terminal.
I tried and its unable to identify openzeppelin files.

**Comment by 0v1se** [comment]
@jchittoda compiling with solc is another story, it doesn't work out of the box. contract reference resolution + file resolution works.

@mbolotov could you implement correct solc compilation when contracts are in node_modules or ethpm folders?

**Comment by jchittoda** [comment]
@0v1se I agree and understand that it will not pick-up directly from node_modules, As it has to be coded in the plugin to have it supported.

As the remix started supporting to pick-up files from node_modules. So thought that it should also be supported in plugin. I use the intellij solidity plugin for my projects and I appreciate your efforts in making this.

**Comment by mbolotov** [comment]
@jchittoda, we've introduced the 'solc additional parameters' field in the Solidity settings.
So you can use the solc path redirects feature using "prefix=path" pattern. I successfully compiled your sample in that way.
@all : As for the direct support for 'node_modules' folder - do you think it's worth to support such a specific folder? Is it the only one that could be needed for all?

**Comment by 0v1se** [comment]
@mbolotov truffle seems to be de facto standard for solidity smart contract development tool
truflle support packages from npm and ethpm:

https://truffleframework.com/docs/truffle/getting-started/package-management-via-ethpm
https://truffleframework.com/docs/truffle/getting-started/package-management-via-npm

both these are supported by the plugin (in terms of file/contract resolution)
see https://github.com/intellij-solidity/intellij-solidity/blob/e478cff85eb636a78530b980b89bc24447f59ddc/src/main/kotlin/me/serce/solidity/lang/resolve/ref/SolImportPathReference.kt

**Comment by mbolotov** [comment]
@0v1se 
Okay, I'll look into this
